src/compiler/parser.rs

```diff
@@ -161,42 +159,42 @@ impl ReadonlyTextRange for MissingNode {
 
 #[allow(non_snake_case)]
 struct ParserType {
-    scanner: RwLock<Scanner>,
+    scanner: RefCell<Scanner>,
     NodeConstructor: Option<fn(SyntaxKind, usize, usize) -> BaseNode>,
     IdentifierConstructor: Option<fn(SyntaxKind, usize, usize) -> BaseNode>,
     TokenConstructor: Option<fn(SyntaxKind, usize, usize) -> BaseNode>,
     SourceFileConstructor: Option<fn(SyntaxKind, usize, usize) -> BaseNode>,
     factory: NodeFactory,
     file_name: Option<String>,
-    parse_diagnostics: Option<RwLock<Vec<DiagnosticWithDetachedLocation>>>,
-    current_token: RwLock<Option<SyntaxKind>>,
+    parse_diagnostics: Option<RefCell<Vec<DiagnosticWithDetachedLocation>>>,
+    current_token: RefCell<Option<SyntaxKind>>,
     parsing_context: Option<ParsingContext>,
-    parse_error_before_next_finished_node: AtomicBool,
+    parse_error_before_next_finished_node: Cell<bool>,
```

Comment:
I guess you can use `Cell`'s instead of `RefCell`'s when the wrapped data is `Copy`